charmcraft/commands/store/__init__.py

```diff
@@ -560,6 +560,43 @@ def run(self, parsed_args):
         logger.info(msg, *args)
 
 
+class CloseCommand(BaseCommand):
+    """Close a channel for a charm or bundle."""
+
+    name = "close"
+    help_msg = "Close a channel for a charm or bundle"
+    overview = textwrap.dedent(
+        """
+        Close the specified channel for a charm or bundle.
+
+        The channel is made up of `track/risk/branch` with both the track and
+        the branch as optional items, so formally:
+
+          [track/]risk[/branch]
+
+        Channel risk must be one of stable, candidate, beta or edge. The
+        track defaults to `latest` and branch has no default.
+
+        Closing a channel will take you through login if needed.
```

Comment:
we will need to update this if we go down the agreed path of saying "Not logged in" if the user has never logged in before.

Reply:
IIUC we should ask automatically for login in all operations except the auth-related ones (like `whoami`).